File: VB/GrowthWare.Framework/Model/Profiles/MClientChoices.vb
﻿Namespace Model.Profiles
    ''' <summary>
    ''' All propertyies represent the column names in the ZF_ACCT_CHOICES table.
    ''' </summary>
    ''' <remarks></remarks>
    <Serializable(), CLSCompliant(True)> _
    Public NotInheritable Class MClientChoices

        ''' <summary>
        ''' Added to prevent the compiler from creating a default constructor.
        ''' Resolves CA 1053
        ''' </summary>
        ''' <remarks></remarks>
        Private Sub New()

        End Sub

        ''' <summary>
        ''' Gets the records per page.
        ''' </summary>
        ''' <value>The records per page.</value>
        Public Shared ReadOnly Property RecordsPerPage() As String
            Get
                Return "RECORDS_PER_PAGE"
            End Get
        End Property

        ''' <summary>
        ''' Gets the state of the anonymous client choices.
        ''' </summary>
        ''' <value>The state of the anonymous client choices.</value>
        Public Shared ReadOnly Property AnonymousClientChoicesState() As String
            Get
                Return "AnonymousClientChoicesState"
            End Get
        End Property

        ''' <summary>
        ''' Gets the name of the session.
        ''' </summary>
        ''' <value>The name of the session.</value>
        Public Shared ReadOnly Property SessionName() As String
            Get
                Return "ClientChoicesState"
            End Get
        End Property

        ''' <summary>
        ''' Gets the name of the account.
        ''' </summary>
        ''' <value>The name of the account.</value>
        Public Shared ReadOnly Property AccountName() As String
            Get
                Return "ACCT"
            End Get
        End Property

        ''' <summary>
        ''' Gets the security entity ID.
        ''' </summary>
        ''' <value>The security entity ID.</value>
        Public Shared ReadOnly Property SecurityEntityId() As String
            Get
                Return "SE_SEQ_ID"
            End Get
        End Property

        ''' <summary>
        ''' Gets the name of the security entity.
        ''' </summary>
        ''' <value>The name of the security entity.</value>
        Public Shared ReadOnly Property SecurityEntityName() As String
            Get
                Return "SE_NAME"
            End Get
        End Property

        ''' <summary>
        ''' Gets the color of the back.
        ''' </summary>
        ''' <value>The color of the back.</value>
        Public Shared ReadOnly Property BackColor() As String
            Get
                Return "BACK_COLOR"
            End Get
        End Property

        ''' <summary>
        ''' Gets the color of the left.
        ''' </summary>
        ''' <value>The color of the left.</value>
        Public Shared ReadOnly Property LeftColor() As String
            Get
                Return "LEFT_COLOR"
            End Get
        End Property

        ''' <summary>
        ''' Gets the color of the head.
        ''' </summary>
        ''' <value>The color of the head.</value>
        Public Shared ReadOnly Property HeadColor() As String
            Get
                Return "HEAD_COLOR"
            End Get
        End Property

        ''' <summary>
        ''' Gets the color of the header fore.
        ''' </summary>
        ''' <value>The color of the header fore.</value>
        Public Shared ReadOnly Property HeaderForeColor() As String
            Get
                Return "Header_ForeColor"
            End Get
        End Property

        ''' <summary>
        ''' Gets the color of the subhead.
        ''' </summary>
        ''' <value>The color of the subhead.</value>
        Public Shared ReadOnly Property SubheadColor() As String
            Get
                Return "SUB_HEAD_COLOR"
            End Get
        End Property

        ''' <summary>
        ''' Gets the color of the row back.
        ''' </summary>
        ''' <value>The color of the row back.</value>
        Public Shared ReadOnly Property RowBackColor() As String
            Get
                Return "Row_BackColor"
            End Get
        End Property

        ''' <summary>
        ''' Gets the color of the alternating row back.
        ''' </summary>
        ''' <value>The color of the alternating row back.</value>
        Public Shared ReadOnly Property AlternatingRowBackColor As String
            Get
                Return "AlternatingRow_BackColor"
            End Get
        End Property

        ''' <summary>
        ''' Gets the color scheme.
        ''' </summary>
        ''' <value>The color scheme.</value>
        Public Shared ReadOnly Property ColorScheme() As String
            Get
                Return "COLOR_SCHEME"
            End Get
        End Property

        ''' <summary>
        ''' Gets the action.
        ''' </summary>
        ''' <value>The action.</value>
        Public Shared ReadOnly Property Action() As String
            Get
                Return "Favorite_Action"
            End Get
        End Property

    End Class
End Namespace

File: CS/GrowthWare.Framework/Model/Profiles/MClientChoices.cs
﻿using System;

namespace GrowthWare.Framework.Model.Profiles
{
    /// <summary>
    /// Class MClientChoices
    /// </summary>
    [Serializable(), CLSCompliant(true)]
    public sealed class MClientChoices
    {

        private MClientChoices()
        {
        }

        /// <summary>
        /// Gets the records per page.
        /// </summary>
        /// <value>The records per page.</value>
        public static string RecordsPerPage
        {
            get { return "RECORDS_PER_PAGE"; }
        }

        /// <summary>
        /// Gets the state of the anonymous client choices.
        /// </summary>
        /// <value>The state of the anonymous client choices.</value>
        public static string AnonymousClientChoicesState
        {
            get { return "AnonymousClientChoicesState"; }
        }

        /// <summary>
        /// Gets the name of the session.
        /// </summary>
        /// <value>The name of the session.</value>
        public static string SessionName
        {
            get { return "ClientChoicesState"; }
        }

        /// <summary>
        /// Gets the name of the account.
        /// </summary>
        /// <value>The name of the account.</value>
        public static string AccountName
        {
            get { return "ACCT"; }
        }

        /// <summary>
        /// Gets the security entity ID.
        /// </summary>
        /// <value>The security entity ID.</value>
        public static string SecurityEntityId
        {
            get { return "SE_SEQ_ID"; }
        }

        /// <summary>
        /// Gets the name of the security entity.
        /// </summary>
        /// <value>The name of the security entity.</value>
        public static string SecurityEntityName
        {
            get { return "SE_NAME"; }
        }

        /// <summary>
        /// Gets the color of the back.
        /// </summary>
        /// <value>The color of the back.</value>
        public static string BackColor
        {
            get { return "BACK_COLOR"; }
        }

        /// <summary>
        /// Gets the color of the left.
        /// </summary>
        /// <value>The color of the left.</value>
        public static string LeftColor
        {
            get { return "LEFT_COLOR"; }
        }

        /// <summary>
        /// Gets the color of the head.
        /// </summary>
        /// <value>The color of the head.</value>
        public static string HeadColor
        {
            get { return "HEAD_COLOR"; }
        }

        /// <summary>
        /// Gets the color of the header fore.
        /// </summary>
        /// <value>The color of the header fore.</value>
        public static string HeaderForeColor
        {
            get { return "Header_ForeColor"; }
        }

        /// <summary>
        /// Gets the color of the subhead.
        /// </summary>
        /// <value>The color of the subhead.</value>
        public static string SubheadColor
        {
            get { return "SUB_HEAD_COLOR"; }
        }

        /// <summary>
        /// Gets the color of the row back.
        /// </summary>
        /// <value>The color of the row back.</value>
        public static string RowBackColor
        {
            get { return "Row_BackColor"; }
        }

        /// <summary>
        /// Gets the color of the alternating row back.
        /// </summary>
        /// <value>The color of the alternating row back.</value>
        public static string AlternatingRowBackColor
        {
            get { return "AlternatingRow_BackColor"; }
        }

        /// <summary>
        /// Gets the color scheme.
        /// </summary>
        /// <value>The color scheme.</value>
        public static string ColorScheme
        {
            get { return "COLOR_SCHEME"; }
        }

        /// <summary>
        /// Gets the action.
        /// </summary>
        /// <value>The action.</value>
        public static string Action
        {
            get { return "Favorite_Action"; }
        }
    }

}
